search/src/main/java/com/jjswigut/search/ui/SearchFragment.kt

```diff
@@ -0,0 +1,103 @@
+    private val binding get() = _binding!!
+
+    private val searchViewModel: SearchViewModel by activityViewModels()
+
+    private lateinit var fusedLocationClient: FusedLocationProviderClient
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(requireActivity())
+        locationHelper.getLocationPermission(requireContext(), requireActivity())
+
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentSearchBinding.inflate(inflater, container, false)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.chineseButton.setOnClickListener {
+            searchViewModel.foodSelection.value = getString(R.string.chinese)
+        }
+        binding.italianButton.setOnClickListener {
+            searchViewModel.foodSelection.value = getString(R.string.italian)
+        }
+        binding.bbqButton.setOnClickListener {
+            searchViewModel.foodSelection.value = getString(R.string.bbq)
+        }
+        binding.americanButton.setOnClickListener {
+            searchViewModel.foodSelection.value = getString(R.string.american)
+        }
+        binding.coffeeButton.setOnClickListener {
+            searchViewModel.foodSelection.value = getString(R.string.coffee)
+        }
+        binding.deliButton.setOnClickListener {
+            searchViewModel.foodSelection.value = getString(R.string.deli)
+        }
+        binding.breakfastButton.setOnClickListener {
+            searchViewModel.foodSelection.value = getString(R.string.breakfast)
+        }
+
+        binding.radiusSelector.addOnChangeListener(Slider.OnChangeListener { slider, value, fromUser ->
+            searchViewModel.searchRadius.value = slider.value.toInt()
+        })
+
+        searchViewModel.searchRadius.observe(viewLifecycleOwner,
+            { binding.radius.text = getString(R.string.radius, it) })
+
+        searchViewModel.foodSelection.observe(viewLifecycleOwner, { binding.foodType.text = it })
+
+
+        binding.searchButton.setOnClickListener {
+            if (locationHelper.userLocation != null) {
+                searchViewModel.navigateToCardStackWithQueryParams(
+                    view,
+                    locationHelper.userLocation!!.latitude.toFloat(),
+                    locationHelper.userLocation!!.longitude.toFloat()
+                )
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        locationHelper.getLastLocation(fusedLocationClient, requireContext(), requireActivity())
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        _binding = null
```

Comment:
probably want this in on destroy view - probably also want to try to tie location updates to this fragment's lifecycle if this is where you're using it